apps/idos-enclave/src/pages/methods/Password.tsx

```diff
@@ -1,43 +1,70 @@
+import * as Base64Codec from "@stablelib/base64";
 import { useCallback, useEffect, useRef, useState } from "preact/hooks";
+import nacl from "tweetnacl";
 import { Button } from "../../components/Button";
 import { Heading } from "../../components/Heading";
 import { Input } from "../../components/Input";
 import { Paragraph } from "../../components/Paragraph";
+import { idOSKeyDerivation } from "../../lib/idOSKeyDerivation";
 import type { MethodProps } from "./Chooser";
 
 export default function Password({
   mode,
   onSuccess,
   store,
-}: MethodProps<{ password: string; duration: number }>) {
+  encryptionPublicKey,
+}: MethodProps<{ password: string; duration: number }> & { encryptionPublicKey: string }) {
   const [password, setPassword] = useState("");
   const [duration, setDuration] = useState(7);
   const passwordInput = useRef<{ focus: () => void }>(null);
+  const [error, setError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  async function derivePublicKeyFromPassword(password: string) {
+    const salt = store.get("human-id");
+    const secretKey = await idOSKeyDerivation({ password, salt });
+    const keyPair = nacl.box.keyPair.fromSecretKey(secretKey);
+    return Base64Codec.encode(keyPair.publicKey);
+  }
 
   useEffect(() => {
     passwordInput.current?.focus();
   }, [passwordInput]);
 
   const onSubmit = useCallback(
-    (e: Event) => {
+    async (e: Event) => {
       e.preventDefault();
       e.stopPropagation();
-
+      setIsLoading(true);
+
+      const derivedPK = await derivePublicKeyFromPassword(password);
+      if (derivedPK !== encryptionPublicKey) {
```

Comment:
Please make sure this comparison is only done when a user has a `current_public_key`, otherwise it will fail for new users. Same thing regarding existing users for whom `current_public_key` will be empty, unless Alex backfilled the database for everyone.

Reply:
All users must have `current_public_key`. 
https://github.com/idos-network/idos-schema/blob/master/schema.production.kf#L29
```sql
table humans {
    id uuid primary,
    current_public_key text notnull,
    inserter text notnull
}
```

Reply:
@sapience can you shed a little light on this backfill please?